connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -759,3 +764,97 @@ async function updateParentsField({
     parents,
   });
 }
+
+export async function microsoftDeletionActivity({
+  connectorId,
+  nodeIdsToDelete,
+}: {
+  connectorId: ModelId;
+  nodeIdsToDelete: string[];
+}): Promise<Result<void, Error>> {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector ${connectorId} not found`));
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  async function deleteNodeAndChildren(
+    nodeId: string
+  ): Promise<Result<void, Error>> {
+    const node = await MicrosoftNodeResource.fetchByInternalId(
+      connectorId,
+      nodeId
+    );
+
+    if (!node) {
+      logger.warn({ connectorId, nodeId }, "Node not found for deletion");
+      return new Ok(undefined);
+    }
+
+    const { nodeType } = typeAndPathFromInternalId(nodeId);
+
+    if (nodeType === "file") {
+      try {
+        await deleteFromDataSource(dataSourceConfig, node.internalId);
+      } catch (error) {
+        logger.error(
+          { connectorId, nodeId, error },
+          `Failed to delete document ${node.internalId} from core data source`
+        );
+        return new Err(
+          new Error(`Failed to delete document ${node.internalId}`)
+        );
+      }
+    } else if (nodeType === "folder" || nodeType === "drive") {
+      const children = await node.fetchChildren();
```

Comment:
there's a pagination question with fetchChildren; IMHO it just fetches the 200 first children... we should change that to use getAllPaginatedEntities
and add a task as follow up to handle better

Reply:
handling better probably meaning we should have a paginationable fetchchildren, enabling multiple small-sized activities for each page as in sync files (better than 1 big activity)